tests/unit/task/task_builder_test.py

```diff
@@ -0,0 +1,190 @@
+
+        assert job.variables.pop("x") == 1
+
+    def test_complete_job_called(self, original_task_function: Callable, task_config: TaskConfig,
+                                 mocked_job_with_adapter: Job):
+        task_builder.build_job_handler(original_task_function, task_config)(mocked_job_with_adapter)
+        mocked_job_with_adapter.set_success_status.assert_called_once()
+
+    def test_returned_task_runs_original_function(self, original_task_function: Callable, task_config: TaskConfig,
+                                                  mocked_job_with_adapter: Job):
+        task_builder.build_job_handler(original_task_function, task_config)(mocked_job_with_adapter)
+
+        original_task_function.assert_called_once()
+
+    def test_before_decorator_called(self, original_task_function: Callable, decorator: TaskDecorator,
+                                     task_config: TaskConfig,
+                                     mocked_job_with_adapter: Job):
+        task_config.before.append(decorator)
+
+        task_builder.build_job_handler(original_task_function, task_config)(mocked_job_with_adapter)
+
+        task_config.before.pop().assert_called_once()
+
+    def test_after_decorator_called(self, original_task_function: Callable, decorator: TaskDecorator,
+                                    task_config: TaskConfig,
+                                    mocked_job_with_adapter: Job):
+        task_config.after.append(decorator)
+
+        task_builder.build_job_handler(original_task_function, task_config)(mocked_job_with_adapter)
+
+        task_config.after.pop().assert_called_once()
+
+    def test_failing_decorator_continues(self, original_task_function: Callable, decorator: TaskDecorator,
+                                         task_config: TaskConfig, mocked_job_with_adapter: Job):
+        decorator.side_effect = Exception()
+        task_config.before.append(decorator)
+
+        # Assert no exception is raised
+        task_builder.build_job_handler(original_task_function, task_config)(mocked_job_with_adapter)
+        decorator.assert_called_once()
+        task_config.exception_handler.assert_not_called()
+
+    def test_decorator_variables_are_added(self, original_task_function: Callable, decorator: TaskDecorator,
+                                           task_config: TaskConfig, mocked_job_with_adapter: Job):
+        decorator_return_value = mocked_job_with_adapter
+        decorator_return_value.variables = {"x": 2}
+        decorator.return_value = decorator_return_value
+
+        job = task_builder.build_job_handler(original_task_function, task_config)(mocked_job_with_adapter)
+
+        assert "x" in job.variables
+
+
+class TestConvertToDictFunction:
+    def test_converting_to_dict(self):
+        dict_function = task_builder.convert_to_dict_function(lambda x: x, "x")
+
+        assert {"x": 1} == dict_function(1)
+
+
+class TestGetFunctionParameters:
```

Comment:
I think all tests could be rewritten to one, with pytest parameterization. Maybe move the dummy functions to a separate module, then something like

```py
@pytest.mark.parametrize("fn,expected", [
    (dummy_functions.no_param, []), 
    (dummy_functions.lambda_fn_one_param, ["x"]), 
    (dummy_functions.mul_pos_func, ["x", "y", "z"]), # and so on... 
])
def test_get_no_param(fn, expected):
    assert task_builder.get_parameters_from_function(fn) == expected
```